pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -421,7 +421,7 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 					},
 					{
 						Alert: "OrphanedVirtualMachineInstances",
-						Expr:  intstr.FromString("((count by (node) (kube_pod_status_ready{condition='true',pod=~'virt-handler.*'} * on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'})) or (count by (node)(kube_pod_info{pod=~'virt-launcher.*'})*0)) == 0"),
+						Expr:  intstr.FromString("(sum by (node) (kube_pod_status_ready{condition='true',pod=~'virt-handler.*'} * on(pod) group_left(node) (sum(kube_pod_info{pod=~'virt-handler.*'}) by (node, pod) )or (count by (node)(sum(kube_pod_info{pod=~'virt-launcher.*', node != ''} ) by (node))*0))) + on(node) group_left(_blah) (sum(kube_pod_info{pod=~'virt-launcher.*', node != ''} *0 ) by (node)) == 0"),
```

Comment:
Regarding the query, can you please elaborate on the following:
* what is the purpose of using `*0`?
* what is the desired effect of using `group_left(_blah)`?

Reply:
> Regarding the query, can you please elaborate on the following:
> 
> * What is the purpose of using `*0`?
The PromQL query is a mathematical calculation on the value. The *0 is intended so that we join on the metric but it will not affect the end value
> * What is the desired effect of using `group_left(_blah)`?
The _blah is a convention that we use instead of leaving this empty. It means that we dont return any value only joining on the filed after the "on" and filter the results based on the metric we join on.

Reply:
@sradco Thanks for the explanation! Can we align the `by` expression closer to the `sum` or `count` i.e. `sum by...` and `count by...` I think it will be more readable

Reply:
For future reviewers:
I suggested a different  query (a bit simpler):

((sum by (node) (kube_pod_status_ready{condition='true',pod=~'virt-handler.*'} * on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'}) ) == 1) or (count by (node)( kube_pod_info{pod=~'virt-launcher.*'})*0) == 0
Explanation:

(sum by (node) (kube_pod_status_ready{condition='true',pod=~'virt-handler.*'} * on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'}) ) == 1)
this query will return metric for each node with healthy virt-handler
so the result will be something like :
{node="node01"} 1
{node="node02"} 1
for nodes with virt-hanlder which is running and healthy

(count by (node)( kube_pod_info{pod=~'virt-launcher.*'})*0)
this query will return nodes with virt-launcher running and value of 0 (always)
so the result will be something like
{node="node01"} 0
{node="node02"} 0
{node="node03"} 0

or between both queries will result metrics with value of 1 where virt-handler is running and heathy
something like this(in the example) :

{node="node01"} 1
{node="node02"} 1
{node="node03"} 0

Reply:
@Barakmor1 That would be great, the problem is that the `kube_pod_status_ready` metric doesn't have the `node` information, that is why we need to multiply it by some other metric vector that does have this `node` information. So `sum by (node)` won't work here

Reply:
@Barakmor1 Perhaps we can open an issue to [`kube-state-metrics`](https://github.com/kubernetes/kube-state-metrics/blob/master/docs/pod-metrics.md) to include the `node` information in the `kube_pod_status_ready` metric.

Reply:
@Barakmor1 Sorry, my bad. I didn't notice that you are already doing the multiplication by the `kube_pod_info`. Please ignore my previous comments